src/main/java/com/scottlogic/librarygradproject/Services/BorrowService.java

```diff
@@ -73,27 +73,19 @@ public void bookReturned(long borrowId) {
             throw new BookAlreadyReturnedException(borrowId);
         }
         borrowToReturn.setActive(false);
+        borrowToReturn.setReturnDate(LocalDate.now());
         long bookId = borrowToReturn.getBookId();
         if (isBorrowed(bookId)) {
             throw new BookAlreadyBorrowedException(bookId);
         }
-        List<Reservation> reservations = reservationRepository.findAllByBookIdOrderByQueuePositionAsc(bookId);
-        if (!reservations.isEmpty()) {
-            Reservation firstReservation = reservations.remove(0);
-            borrowRepository.save(Borrow.builder()
-                    .bookId(bookId)
-                    .userId(firstReservation.getUserId())
-                    .isActive(true)
-                    .borrowDate(LocalDate.now())
-                    .returnDate(LocalDate.now().plusDays(7))
-                    .build());
-            reservationRepository.delete(firstReservation);
+        Reservation reservationToCollect = reservationRepository.findOneByBookIdAndQueuePosition(bookId, 1);
+        if (reservationToCollect != null) {
+            reservationToCollect.setCollectBy(LocalDate.now().plusDays(3));
         }
-        final LongWrapper queuePosition = new LongWrapper(1);
-        reservations.forEach(reservation -> {
-                   reservation.setQueuePosition(queuePosition.getValue());
-                   queuePosition.increment();
-        });
+    }
+
+    public void bookCollected(long bookId) {
+
```

Comment:
Why do we have this empty function?

Reply:
Because we started it then realised we wanted to push what we'd already done so we used git patch to not push those lines but not perfectly